errbot/core_plugins/plugins.py

```diff
@@ -69,13 +70,33 @@ def repos_uninstall(self, mess, args):
     def repos(self, mess, args):
         """ list the current active plugin repositories
         """
+
         installed_repos = self._bot.get_installed_plugin_repos()
+
         all_names = sorted(set([name for name in KNOWN_PUBLIC_REPOS] + [name for name in installed_repos]))
-        return {'repos': [
-            (repo_name in installed_repos, repo_name in KNOWN_PUBLIC_REPOS, repo_name,
-             KNOWN_PUBLIC_REPOS[repo_name][1]
-             if repo_name in KNOWN_PUBLIC_REPOS else installed_repos[repo_name])
-            for repo_name in all_names]}
+
+        repos = {'repos': []}
+
+        for repo_name in all_names:
+
+            installed = False
+            public = False
+
+            try:
+                description = KNOWN_PUBLIC_REPOS[repo_name].get('documentation', 'Unavailable')
+            except KeyError:
+                description = installed_repos[repo_name].get('path')
+
+            if repo_name in KNOWN_PUBLIC_REPOS:
```

Comment:
I'm not sure what you are referring to here? Are you suggesting those lines could be simplified with that single `public = repo_name in KNOWN_PUBLIC_REPOS` line?